requirements/system/functional/server.tex

```diff
@@ -0,0 +1,29 @@
+\subsection{Central Server requirements}
+\subsubsection{Main features}
+\begin{enumerate}
+  \item[FRCS10] The central server shall distribute \glspl{job} among the available work machines using the Scheduling algorithm.
+  \item[FRCS20] Central server shall provide network APIs to:
+  \begin{enumerate}
+    \item[FRCS21] Schedule new jobs for execution.
+    \item[FRCS22] Query status of a given job:
+    \item[FRCS23] Get SSH access to a job's container, provided that:
+    \begin{itemize}
+      \item The \gls{job status} is in Running state, or in Done, but the corresponding docker image has not been deleted yet.
+      \item The user doing this request is the same as the user who submitted the job.
+    \end{itemize}
+    \item[FRCS24] Get statistics about workload of the connected work machines.
+    \item[FRCS25] Get statistics about the jobs of a specific user.
+    \item[FRCS26] Get statistics about jobs running which have been running in the past X hours.
+  \end{enumerate}
+\end{enumerate}
+\subsubsection{Scheduling algorithm}
+\begin{enumerate}
+  \item[FRCS30] Jobs shall be executed in the order of their \gls{job priority}: first Urgent, then High, then Normal, and then Low
+    \item[FRCS40] When a new job with priority Urgent is scheduled, any number of jobs with lower priority shall be freed so that
+        execution of the new job can start.
+    \item[FRCS50] When a new job with priority High, Normal or Low is scheduled, the new job shall be added to a list of non-Urgent jobs.
```

Comment:
I think we should add a requirement which states that jobs with low priority will always be paused to allow for the execution of other jobs.